### ISSUE TITLE
How to cancel definition of highlight group?

### ISSUE THREAD

**Issue by fjchen7** [chentoast/marks.nvim#85]
I use [Gitsigns](https://github.com/lewis6991/gitsigns.nvim) and find that color of marks.nvim distracted.
![image](https://user-images.githubusercontent.com/10106636/200993396-a3d824f1-f92b-48a3-9317-249a5a4f27da.png)

 I try to clear color of the highlight group `MarkSignNumHL`.

Here is the effect after setting `:highlight link MarkSignNumHL LineNr` to make color of `MarkSignNumHL` the same with normal line.
![image](https://user-images.githubusercontent.com/10106636/200992435-b0a3ceea-46e4-449f-bcb8-3fe7c36ff8c4.png)

But this is not quite what I want. I would like to make color `MarkSignNumHL` **transparent**, to make it not overwrite color of other highlight definication. That is to say, in the screen above all line number of marks follows the color of Gitsigns (all green defined by `GitSignsAdd ` in this case).

How can I do this? Thanks a lot!

**Comment by chentoast** [comment]
It sounds like what you want is to set `bg=NONE` in the definition of your highlight group - does that solve the issue?

**Comment by fjchen7** [comment]
No. I try the command `hi MarkSignNumHL guibg=NONE itermbg=NONE`, but it doesn't work. The effect is just like the picture above shows.

**Comment by fjchen7** [comment]
`marks.nvim` refresh both Sign and NR column when adding a new mark. Even though I remove highlight by `:hi! clear MarkSignNumHL`, the highlights are still here (may fallback to a default color?).

I think you should separate Sign and NR toggle command, like `MarksToggleSign` and `MarksToggleNr`. Currently only a `MarksToggle` toggles both together.

**Comment by wustho** [comment]
Same here, this was too distracting...

**Comment by jthvai** [comment]
One workaround is to set gitsigns.nvim's `opts.sign_priority = 11` (or any value higher than marks, defaulting at 10), or to set marks.nvim's `opts.sign_priority = 5` (or any value lower than gitsigns, defaulting at 6). This will allow gitsigns to override the numhl set by marks.

**Comment by jimzk** [comment]
I share my workaround here.

```lua
return {
  "chentoast/marks.nvim",
  config = function(_, opts)
    local marks_utils = require("marks.utils")
    -- Tweak from https://github.com/chentoast/marks.nvim/blob/a69253e4b471a2421f9411bc5bba127eef878dc0/lua/marks/utils.lua#L9
    marks_utils.add_sign = function(bufnr, text, line, id, group, priority)
      priority = priority or 10
      local sign_name = "Marks_" .. text
      if not marks_utils.sign_cache[sign_name] then
        marks_utils.sign_cache[sign_name] = true
        vim.fn.sign_define(sign_name, { text = text, texthl = "MarkSignHL" })
      end
      vim.fn.sign_place(id, group, sign_name, bufnr, { lnum = line, priority = priority })
    end

    require("marks").setup(opts)
  end,
}
```
